examples/observability/adot-amp-grafana-for-haproxy/main.tf

```diff
@@ -98,6 +97,32 @@ module "eks_blueprints_kubernetes_addons" {
 # Observability Resources
 #---------------------------------------------------------------
 
+module "managed_grafana" {
+  source  = "terraform-aws-modules/managed-service-grafana/aws"
```

Comment:
<img width="483" alt="image" src="https://user-images.githubusercontent.com/10175027/174857693-003c4f4d-58dd-4c38-82bf-058af8b52058.png">

Reply:
The same can be said for resources like VPC - it is highly unlikely that users will use the VPC provided in the examples, they will most likely already have their own VPC in their environment(s) and will not provision a new VPC since the example provides one. Its just there to support the user experience to quickly deploy all of the resources and see the outcome

Reply:
I agree with your point of examples being as straightforward as possible. At the same time, there's a portion of customer using their own Grafana for example for many reasons which we should consider. I do not see that comparable with the VPC but more with managed ADOT add-on vs. self-managed otel operator.
Plus showing both steps, with AMG by default for example, highlights the simplicity of using AMG.

Reply:
Understood - but now what we are talking about is a software deliverable, an artifact, which in Terraform speak is a module. Currently in blueprints, an example is not a module - its a demonstration of how to use blueprints to achieve an outcome.

Reply:
Side question, do you see the current example evolving to packaged modules?

Reply:
@bryantbiggs what if we only document that it is possible to reuse an existing workspace? (which is possible). That way we can support both scenarios. And with the pricing model of AMG, customers won't be charged if they are not using the auto-created one

Reply:
I agree on having both options(existing and new) for AMP and AMG

Reply:
I've added back in the `grafana_endpoint` variable and updated the logic to not create the AMG workspace if an endpoint is provided in https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/664/commits/b5a702e760f7829661ad20965c8badf91fc1ab80

However, I do want to stress that examples are not dynamic and variables should only be used for values that MUST be sourced from users (i.e. - the Grafana API key because we currently do not have a means to generate that currently). Whether users copy+paste or git clone, either route provides them direct access to modify the example to suite their environment/intentions. For example - to enable/disable/configure additional addons, use an existing VPC, etc. When we start to add variables it signals to users that this is now intended to be used in situ as if it is a module which is not the intended use case here

Reply:
scratch that - I thought I had a solution but since this is being fed into the provider block it has to be evaluated early on in the process and  conditional logic won't work.

Reply:
Hey @bryantbiggs, are you ok to merge this? I think having the `grafana_endpoint` is enough. I can add a oneliner doc to say they can reuse an existing workspace if they do not wish to use the provided one

Reply:
@bonclay7 yes - I just wanted to be mindful of what you all have created and try to reach a resolution (I think I have exhausted those currently at the moment) before proceeding. But if you are ok to move forward, I'd be happy to merge